actionners/kubernetes/drain/drain.go

```diff
@@ -0,0 +1,257 @@
+	})
+
+	for _, p := range pods.Items {
+
+		if parameters["ignore_daemonsets"] != nil || parameters["ignore_statefulsets"] != nil || parameters["min_healthy_replicas"] != nil {
+			pod, err := client.GetPod(p.Name, p.Namespace)
+			if err != nil {
+				return utils.LogLine{
+						Objects: objects,
+						Error:   err.Error(),
+						Status:  "failure",
+					},
+					err
+			}
+
+			if len(pod.OwnerReferences) != 0 {
+				switch pod.OwnerReferences[0].Kind {
+				case "DaemonSet":
+					if parameters["ignore_daemonsets"].(bool) {
+						return utils.LogLine{
+								Objects: objects,
+								Result:  fmt.Sprintf("the pod %v in the namespace %v belongs to a Daemonset and ignore_daemonsets is true", p.Name, p.Namespace),
+								Status:  "ignored",
+							},
+							nil
+					}
+				case "StatefulSet":
+					if parameters["ignore_statefulsets"].(bool) {
+						return utils.LogLine{
+								Objects: objects,
+								Result:  fmt.Sprintf("the pod %v in the namespace %v belongs to a Statefulset and ignore_statefulsets is true", p.Name, p.Namespace),
+								Status:  "ignored",
+							},
+							nil
+					}
+				case "ReplicaSet":
+					if parameters["min_healthy_replicas"] != nil {
+						u, errG := client.GetReplicasetFromPod(pod)
+						if errG != nil {
+							return utils.LogLine{
+									Objects: objects,
+									Error:   errG.Error(),
+									Status:  "failure",
+								},
+								errG
+						}
+						if u == nil {
+							return utils.LogLine{
+									Objects: objects,
+									Error:   fmt.Sprintf("can't find the replicaset for the pod %v in namespace %v", p.Name, p.Namespace),
+									Status:  "failure",
+								},
+								fmt.Errorf("can't find the replicaset for the pod %v in namespace %v", p.Name, p.Namespace)
+						}
+						if strings.Contains(fmt.Sprintf("%v", parameters["min_healthy_replicas"]), "%") {
+							v, _ := strconv.ParseInt(strings.Split(parameters["min_healthy_replicas"].(string), "%")[0], 10, 64)
+							if v > int64(100*u.Status.ReadyReplicas/u.Status.Replicas) {
+								return utils.LogLine{
+										Objects: objects,
+										Result:  fmt.Sprintf("not enough healthy pods in the replicaset of the pod %v in namespace %v", p.Name, p.Namespace),
+										Status:  "ignored",
+									},
+									fmt.Errorf("not enough healthy pods in the replicaset of the pod %v in namespace %v", p.Name, p.Namespace)
+							}
+						} else {
+							v, _ := strconv.ParseInt(parameters["min_healthy_replicas"].(string), 10, 64)
+							if v > int64(u.Status.ReadyReplicas) {
+								return utils.LogLine{
+										Objects: objects,
+										Result:  fmt.Sprintf("not enough healthy pods in the replicaset of the pod %v in namespace %v", p.Name, p.Namespace),
+										Status:  "ignored",
+									},
+									fmt.Errorf("not enough healthy pods in the replicaset of the pod %v in namespace %v", p.Name, p.Namespace)
+							}
+						}
+					}
+				}
+			}
+		}
```

Comment:
We should transform this into a method because it's the same in 2 different actioners

Reply:
Did this and refactored kubernetes:terminate to leverage it. There are some caveats in terms how we handle it for drain and terminate since drain needs to do it in a loop stile.